### PR TITLE
feat(*): Polish Week - Discord API Types Edition

### PR DESCRIPTION
**Pull request by vladfrangu**
Discord had a polish week, where they went through ALL the API docs PRs and what nots. Now it's time for us to get up to date

**Please describe the changes this PR makes and why it should be merged:**

- Restructures the module (@devsnek suggestion)

**Reference Discord API Docs PRs or commits:**

- 216bfca63d1acfd0d997c82b2adc945d3cd1a8c2 & d315abbcdd268f3916563bc84445c81152801685 -> https://github.com/discord/discord-api-docs/commit/47eab31c044cb2dec9ab36fcf9583cacc4701df5
- dfcf4a1efe2e7b244c094293634ce7d1e7b8b32c -> https://github.com/discord/discord-api-docs/commit/efe4e5808b6826d40302e265a5ae9b5b65d92fe7
- 8694342e94677c0a9980d5c65138a4cc85526de7 -> https://github.com/discord/discord-api-docs/commit/9293f0d490ac6acf9d627e429e5a8131b303b528
- 8540ab7c424ea62fda8e6eb9a718523c9247a754 -> https://github.com/discord/discord-api-docs/commit/5bf598b864fb89262fce07137f68ce6e7e583432
- 03b006a3d88fbc4211a40a3cd361c9731d6cea26 -> https://github.com/discord/discord-api-docs/commit/1b4e363e324eb1f49a47e32cb0108fbe276c8e0e 
- 32b3e1c185b9e345088aa1aecbe3357a9c13a796 -> https://github.com/discord/discord-api-docs/commit/e887382fafd4c4417f7ba62963984f25bcb643f6
- 8966fc570f755c55bf21c28612b657e2aa68db78 -> https://github.com/discord/discord-api-docs/commit/39b254bed1cc396c475e508a3f2bf328815605c9
- b7d6b5ad426d8eb75a8652d1f8d2b84f372c3171 -> https://github.com/discord/discord-api-docs/commit/70e1cce1dcd5f6427717f85c29077d14a63b28a1
- c672ff9d1588b349b14e5f0057814b62b2074cf1 -> https://github.com/discord/discord-api-docs/commit/d0287477a9d9128d6842d4231bc0a82b5598286d
- f61b84d25fa3e0365c0d7d7d03f19a03f252ed61 -> https://github.com/discord/discord-api-docs/commit/6d3aaefb929fdf5fbd5f3031dd245fd1e62c87fb
- a47a475ddcebc0398eb57f3b2622538c68bde6af -> https://github.com/discord/discord-api-docs/commit/ff0c831e424f1bc17dd3cde62da48d5c3d933e88
- a812af521639162c9af70fc7195d1786ad003c98 -> https://github.com/discord/discord-api-docs/commit/2bf90799d55c88a4a98dae6847316410b069c58f
- cd346896fae604afed9a9db762cde016f1eb05b7 -> https://github.com/discord/discord-api-docs/commit/2641d9808f676e7316483d152cdb37ed1168f968
- 0a4106ab810e0cfef9b9977bf9d2c1fc2c00a723 -> https://github.com/discord/discord-api-docs/commit/a96ea527dc8db7726637a1ec2ee6c8063a01bff6
- 7507161c710c67429be84fe4bdc6bd14c90b6e43 -> https://github.com/discord/discord-api-docs/commit/18514cb487ef1889c9d2be29c74dd114ee8c39db
- 89fdfc8cf9e742387b59849742e63a0bca3649be -> https://github.com/discord/discord-api-docs/commit/8de017436d37e56fab14cb8f68f0448a45ebc731
- c41bfedd0b0890074fb9311768a7fbde05ac2fbd -> https://github.com/discord/discord-api-docs/commit/57965033ab4216a0bb853e85d6912531cd5a9981
- 343b220cec98039026e622802a203c7b7e4d3faf -> https://github.com/discord/discord-api-docs/commit/f2ccd2ac6d736b57045750765a55cb29311f96bc
- 877639128a04687adfb54001c5790283defca482 -> https://github.com/discord/discord-api-docs/commit/b761ed03c07e7fce80175947ec01732099b2b5cd
- 6f40d7fbf74813c9dc0cb4a2fefe85769ef9a8da -> https://github.com/discord/discord-api-docs/commit/fa9a0739eeedf3ff380b77b9fdcf1302aa1b0c26
- 0ba84879f49a3a72206e8f89b25ec51432439a13 -> https://github.com/discord/discord-api-docs/commit/f72b084773d4d3989fb19be4fb4d9cf276a1e6b3
- e8e8df8258222b9f2b2cfeeeaef8358e998a39b7 -> https://github.com/discord/discord-api-docs/commit/9f9c28313455ac4578a487a72aa97b2fea4a832f
- 8504a2d7ba915be87b205517ba03534150873ea3 -> https://github.com/discord/discord-api-docs/commit/c7d25885c5cd80a49b31609a40b70603b35f9dec
- 48b25cc143453935d6a850962208f493ada1b1c7 -> https://github.com/discord/discord-api-docs/commit/ab7a51ffea418a354562404da51ee609f8bca539
- 14d0f4a33c1f4cf7a28dfc2c755b3dff7a6e34c5 -> https://github.com/discord/discord-api-docs/commit/f36156dbb641f5c4d4f4593f345bfd6e27fdee08
- 66c58b4ffae9933a4d4d895651e10306de1a439e -> https://github.com/discord/discord-api-docs/commit/eaa12cbc8f96cf7cfe8c530f88e60582c24ca5dd
- b73875fd6da2521ac22e8cc72f1af1d1378f258d -> https://github.com/discord/discord-api-docs/commit/73dfc0ab9bcf87ad8843b0d399de747190f0cb0d

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,9 +32,19 @@ const { APIUser } = require('discord-api-types/v8');
 import { APIUser } from 'discord-api-types/v8';
 ```
 
-You may also import just certain parts of the module that you need. The possible values are: `globals`, `gateway`, `gateway/v*`, `payloads`, `payloads/v*`, `rest`, `rest/v*`, `rpc`, `voice` and `voice/v*`.
+You may also import just certain parts of the module that you need. The possible values are: `globals`, `gateway`, `gateway/v*`, `payloads`, `payloads/v*`, `rest`, `rest/v*`, `rpc`, `utils`, `utils/v*`, `voice` and `voice/v*`.
```
Comment (kyranet) [inline README.md:35]: Oxford comma:
```suggestion
You may also import just certain parts of the module that you need. The possible values are: `globals`, `gateway`, `gateway/v*`, `payloads`, `payloads/v*`, `rest`, `rest/v*`, `rpc`, `utils`, `utils/v*`, `voice`, and `voice/v*`.
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: payloads/v8/oauth2.ts [hunk 2]
```diff
@@ -38,6 +38,14 @@ export interface APIApplication {
 	 * When `true` the app's bot will only join upon completion of the full oauth2 code grant flow
 	 */
 	bot_require_code_grant: boolean;
+	/**
+	 * The url of the app's terms of service
```
Comment (kyranet) [inline payloads/v8/oauth2.ts:42]:
```suggestion
	 * The url of the application's terms of service
```

File: payloads/v8/oauth2.ts [hunk 3]
```diff
@@ -38,6 +38,14 @@ export interface APIApplication {
...
+	 */
+	terms_of_service_url?: string;
+	/**
+	 * The url of the app's privacy policy
```
Comment (kyranet) [inline payloads/v8/oauth2.ts:46]:
```suggestion
	 * The url of the application's privacy policy
```

File: payloads/v8/oauth2.ts [hunk 4]
```diff
@@ -50,7 +58,7 @@ export interface APIApplication {
 	 */
 	summary: string;
 	/**
-	 * The base64 encoded key for the GameSDK's GetTicket
+	 * The hex encoded key for verification in interactions and the GameSDK's GetTicket function
```
Comment (kyranet) [inline payloads/v8/oauth2.ts:61]:
```suggestion
	 * The hexadecimal encoded key for verification in interactions and the GameSDK's GetTicket function
```

File: payloads/v8/webhook.ts [hunk 5]
```diff
@@ -49,8 +49,18 @@ export interface APIWebhook {
...
+	/**
+	 * The guild of the channel that this webhook is following (returned for Channel Follower Webhooks)
+	 */
 	source_guild?: APIPartialGuild;
+	/**
+	 * The channel that this webhook is following (returned for Channel Follower Webhooks)
+	 */
 	source_channel?: APIPartialChannel;
+	/**
+	 * The url used for executing the webhook (returned by the [webhooks](#DOCS_TOPICS_OAUTH2/webhooks) OAuth2 flow)
```
Comment (kyranet) [inline payloads/v8/webhook.ts:61]: Does this even link?

**Review by kyranet — CHANGES_REQUESTED** [review]

File: payloads/v8/webhook.ts (hunk 5 above)
Comment (vladfrangu) [reply payloads/v8/webhook.ts]: No. Fixed in https://github.com/discordjs/discord-api-types/pull/110/commits/e55f73a2a45af60993e22fb46bd90f563af29a4e

**Review by kyranet — APPROVED** [review]

**Pull request merged**